dashboard/lib/services/script_seed.rb

```diff
@@ -285,6 +304,46 @@ def self.import_levels_script_levels(levels_script_levels_data, seed_context)
       destroy_outdated_objects(LevelsScriptLevel, levels_script_levels, levels_script_levels_to_import, seed_context)
     end
 
+    def self.import_resources(resources_data, seed_context)
+      course_version_id = seed_context.script.get_course_version&.id
+      resources_to_import = resources_data.map do |resource_data|
+        resource_attrs = resource_data.except('seeding_key')
+        resource_attrs['course_version_id'] = course_version_id
+        Resource.new(resource_attrs)
+      end
+
+      # Resources are owned by the course version. Therefore, do not delete
```

Comment:
Currently it's impossible to delete a resource using the UI so this isn't a problem now but what would it looks like to allow resource deletion? It seems like it wouldn't interact well with this seeding logic

Reply:
I could imagine this coming up for vocab for instance as there seems to be a fair amount of interest in an "All Vocab" page

Reply:
Destroying the resource should destroy any lessons_resources entries which point to it, like this:
![Screen Shot 2020-12-15 at 2 00 53 PM](https://user-images.githubusercontent.com/8001765/102278241-4965c400-3ede-11eb-9092-fb4d4e904908.png)

That should ensure that we never serialize a bad lessons_resources entry, i.e. one which points to a resources which has been deleted. 

However, as you were probably alluding to, deletion of a resource would never get propagated to another environment. it could get removed from every .script_json file in a course version, but the seed process still wouldn't delete the resource. 

FWIW we have this same problem with scripts and levels in the old (and new) seed processes, i.e. deleting a script or level removes it from the serialized format but does not remove it from other environments. This works ok, because levelbuilders do not have to see this clutter while they are editing. It is not ideal though, because scripts which were deleted in levelbuilder can still be accessed by end users in production. FWIW resources would not have this problem, so this definitely seems like a manageable situation, though still far from ideal.

In order to fix this for resources in particular, it is possible we could start storing resources on the .course_json or .script_json file, depending on whether the script is in a unit or not. then, those resources would become the definitive list, and we could safely delete resources from other environments if they were removed from that list.

Reply:
Thanks for the explanation! Like you mentioned, this is an existing issue in a number of places so I don't think this needs to be solved with this PR -- or maybe ever. I think this warrants more thought _if_ it comes up as I don't love the idea of resources being stored in different files in different situations.

Reply:
> Thanks for the explanation! Like you mentioned, this is an existing issue in a number of places so I don't think this needs to be solved with this PR -- or maybe ever. I think this warrants more thought _if_ it comes up as I don't love the idea of resources being stored in different files in different situations.

this may not happen for some time, but... ideally, the serialized files (.course_json, .script_json) would be refactored to more closely match the new structure of our curriculum, e.g. there should really be something like a .course_version_json file, which either contained unit_json or had one or more .unit_json files that went along with it. then the resources would always be in the .course_version_json. the same file would also hold all the other attributes which are currently shared/duplicated between scripts and courses today, such as family name ,version year, is_stable, pilot_experiment, etc. Given how the new seeding logic is written, I don't think this would require a great deal of re-inventing to accomplish, but it's hard to put it above some of our other upcoming tech debt goals.